Structure_Engine/Query/ReinforcementLayout.cs

```diff
@@ -178,10 +220,100 @@ public static List<Line> ReinforcementLayout(this LongitudinalReinforcement rein
             return rebarLines;
         }
 
+        /***************************************************/
+
+        [Description("Gets the TransverseReinforcement centrelines as a list of polylines, based on inner and outer profile edges and Bar parameters.")]
+        [Input("reinforcement", "The TransverseReinforcement to extract the centrelines from.")]
+        [Input("cover", "Rebar cover.", typeof(Length))]
+        [Input("outerProfileEdges", "The outer profile edges of the ConcreteSection to be populated.")]
+        [Input("innerProfileEdges", "The inner profile edges, or openings, of the ConcreteSection to be populated.")]
+        [Input("length", "Length of the host Bar used to generate the lines", typeof(Length))]
+        [Input("transformation", "Transformation needed to move the lines from the position of the host element.")]
+        [Output("points", "The centrelines of the LongitudinalReinforcement.")]
+        public static List<ICurve> ReinforcementLayout(this TransverseReinforcement reinforcement, double cover, List<ICurve> outerProfileEdges, List<ICurve> innerProfileEdges, double length, TransformMatrix transformation)
+        {
+            List<ICurve> rebarLines = new List<ICurve>();
+            List<ICurve> stirrupOutline = reinforcement.CenterlineLayout.ICurveLayout(outerProfileEdges, innerProfileEdges);
```

Comment:
This should call the ReinforcementLayout method instead of directly calling the curve layout, to make sure any changes there is passed through.